src/expr-test-util/src/lib.rs

```diff
@@ -227,6 +235,38 @@ impl TestDeserializeContext for MirScalarExprDeserializeContext {
         type_name: &str,
         rti: &ReflectedTypeInfo,
     ) -> Option<String> {
+        if type_name == "MirScalarExpr" {
+            let map = json.as_object().unwrap();
+            // Each enum instance only belows to one variant.
+            assert_eq!(map.len(), 1);
+            for (variant, data) in map.iter() {
+                match &variant[..] {
+                    "Column" => return Some(format!("#{}", data.as_u64().unwrap())),
+                    "Literal" => {
+                        let column_type: ColumnType =
+                            serde_json::from_value(data.as_array().unwrap()[1].clone()).unwrap();
+                        match data.as_array().unwrap()[0].as_object().unwrap().get("Ok") {
+                            Some(inner_data) => {
+                                let row: Row = serde_json::from_value(inner_data.clone()).unwrap();
+                                let result = format!(
+                                    "({} {})",
+                                    row.unpack_first(),
+                                    from_json(
+                                        &serde_json::to_value(&column_type.scalar_type).unwrap(),
+                                        "ScalarType",
+                                        rti,
+                                        self
+                                    )
+                                );
+                                return Some(result);
+                            }
+                            None => unreachable!("Literal errors are not supported"),
```

Comment:
This is perhaps worth logging as a separate ticket, since that effectively means that we cannot use this feature for queries involving scalar subqueries.